adagrams/game.py

```diff
@@ -1,13 +1,111 @@
+# comment to practice push to GitHub
 from random import randint
 
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+POINTS = {
+    1:["A", "E", "I", "O", "U", "L", "N", "R", "S", "T"],
+    2:["D","G"],
+    3:["B","C","M","P"],
+    4:["F","H","V","W","Y"],
+    5:["K"],
+    8:["J","X"],
+    10:["Q","Z"]
+    }
+
 def draw_letters():
-    pass
+    total_letter_pool = []
+    hand = []
+
+    # create a list with the total amount of letters: 
+    for letter, quantity in LETTER_POOL.items():
+        total_letter_pool.extend([letter for i in range(quantity)])
+    
+    # draw 10 random letters from list of letters and append it to the hand
+    for i in range(10):
+        random_index = randint(0,len(total_letter_pool)-1)
+        random_letter = total_letter_pool.pop(random_index)
+        hand.append(random_letter)
+    return hand
 
 def uses_available_letters(word, letter_bank):
-    pass
+    input_word = word.upper()
+    copy_letter_bank = letter_bank.copy()
+
+    # Check if each letter in the word exists in the letter bank
+    for char in input_word: 
+        if not char in copy_letter_bank:
+            return False
+        copy_letter_bank.remove(char)
+    return True
 
 def score_word(word):
-    pass
+    total_score = 0
+    input_word = word.upper()
+
+    # Iterate through score values and corresponding letters
+    for score, letter_score in POINTS.items():
+        for char in input_word:
+            if char in letter_score:
+                total_score += score
+    
+    # Bonus points for words with 7 or more letters
+    if len(input_word) >= 7:
+        total_score += 8
```

Comment:
If I had no idea about how to play Adagrams, I'm left wondering "Why do we add 8 to the total_points when the length of the word is greater than 7"

We can make our code more self-documenting by introducing a constant variable, like so:
```py
BONUS_POINTS_FOR_LENGTH = 8
    if len(input_word) >= 7:
        total_score += BONUS_POINTS_FOR_LENGTH